dataplane/kernel/genetlink/genetlink.c

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "genetlink.h"
+
+#include <linux/netlink.h>
+#include <netlink/genl/ctrl.h>
+#include <netlink/genl/family.h>
+#include <netlink/genl/genl.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+enum {
+  /* packet metadata */
+  GENL_PACKET_ATTR_IIFINDEX,
+  GENL_PACKET_ATTR_OIFINDEX,
+  GENL_PACKET_ATTR_CONTEXT,
+  GENL_PACKET_ATTR_DATA,
+};
+
+struct nl_sock* create_port(const char* family, const char* group) {
+  fprintf(stderr, "creating port\n");
+
+  struct nl_sock* sock = nl_socket_alloc();
+  if (sock == NULL) {
+    fprintf(stderr, "error: failed to alloc nl socket");
+    return NULL;
+  }
+  nl_socket_disable_auto_ack(sock);
+  int error = genl_connect(sock);
+  if (error < 0) {
+    fprintf(stderr, "error: failed to disable auto ack");
```

Comment:
Do we want to also print the error code?

Reply:
done